### ISSUE TITLE
Unable to see ray dashboard

### ISSUE THREAD

**Issue by offthewall123** [ray-project/ray#14028]
### What is the problem?

Can't see ray dashboad when visit localhost:8265

### Reproduction (REQUIRED)
Using Ray 0.8.6, one node as head node, another as woker node.
Start successfully.When visiting localhost:8265, the page flash and shows white page, don't know why this happen.

![image](https://user-images.githubusercontent.com/29748042/107476727-66063b80-6bb1-11eb-8b2a-c17f042d2eb2.png)
and i think Ray started right, if refresh the UI tab, can see it for a second then shows white page.

**Comment by rkooo567** [comment]
Hi, @offthewall123. 0.8.6 is a pretty old version, and there's a possibility there was a bug that is already fixed in the later version. Is it possible for you to try 1.1.0? (the latest stable release)?

**Comment by offthewall123** [comment]
> Hi, @offthewall123. 0.8.6 is a pretty old version, and there's a possibility there was a bug that is already fixed in the later version. Is it possible for you to try 1.1.0? (the latest stable release)?

Have tried 1.1.0, but has other issues, can start ray....So i downgrade to 0.8.6...

**Comment by jacobdanovitch** [comment]
I think I'm having the same issue, but can't tell without seeing the devtools log. Figured I'd just post this all here in case it's related/helpful but I can make a new issue for it as well.

* Head node: Random Azure VM (all ports opened)
  * No GPUs
  * Replaced public IP with `${VM_PUBLIC_IP}` below
* Worker node: Desktop at home (`ufw` temporarily disabled)
* Tested from standard CLI using 1.2.0 as well as Docker with 1.2.0 and 1.1.0, matching head and worker.

## Case 1

#### Head Node

Started head node using `ray start --head --node-ip-address=${VM_PUBLIC_IP} --dashboard-host=0.0.0.0`. The dashboard does **not** work, had the same issue described in OP. Popped open devtools:

![image](https://user-images.githubusercontent.com/17260949/107885689-b9d29680-6ec9-11eb-830b-8ad9f34c4415.png)

The relevant lines:

```javascript
// Show GPU features only if there is at least one GPU in cluster.
  const showGPUs =
    nodes.map((n) => n.gpus).filter((gpus) => gpus.length !== 0).length !== 0;
```

#### Worker Node

Ran `ray start --address='${VM_PUBLIC_IP}:6379' --redis-password='...'` and then ran this test script:

```python
import ray
ray.init(address='auto', include_dashboard=False)

@ray.remote
def add(*args): 
    return sum(args)

print(ray.get(add.remote(1, 2)))
```

Error (on the worker node but appears to be a remote traceback from the head node given the IP):

```python
(pid=raylet, ip=10.0.1.4) Traceback (most recent call last):
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 317, in <module>
(pid=raylet, ip=10.0.1.4)     raise e
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 306, in <module>
(pid=raylet, ip=10.0.1.4)     loop.run_until_complete(agent.run())
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
(pid=raylet, ip=10.0.1.4)     return future.result()
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/new_dashboard/agent.py", line 154, in run
(pid=raylet, ip=10.0.1.4)     await site.start()
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/site-packages/aiohttp/web_runner.py", line 128, in start
(pid=raylet, ip=10.0.1.4)     reuse_port=self._reuse_port,
(pid=raylet, ip=10.0.1.4)   File "/home/ray/anaconda3/lib/python3.7/asyncio/base_events.py", line 1389, in create_server
(pid=raylet, ip=10.0.1.4)     % (sa, err.strerror.lower())) from None
(pid=raylet, ip=10.0.1.4) OSError: [Errno 99] error while attempting to bind on address ('${VM_PUBLIC_IP}', 0): cannot assign requested address
Aborted (core dumped)
```

Also tried starting the worker node with `--dashboard-host='0.0.0.0'` but it core dumped as well (no stack trace at all this time). Using `--include-dashboard=false` had no effect.

## Case 2

#### Head Node

Started head node using just `ray start --head --dashboard-host=0.0.0.0`, not providing `--node-ip-address`. Dashboard now **works**.

#### Worker Node

I started the worker the same way as before and then re-ran the test script:

```shell
2021-02-14 11:20:52,433 INFO worker.py:657 -- Connecting to existing Ray cluster at address: ${VM_PUBLIC_IP}
[2021-02-14 11:23:02,726 W 233 233] redis_context.cc:336: Could not allocate Redis context, will retry in 100 milliseconds.
Segmentation fault (core dumped)
```

It hangs for a few minutes and no workers appeared on the dashboard in the meantime.

<hr/>

In summary:

* When I don't use `--node-ip-address`, issues with the dashboard are solved, but I'm unable to connect from a worker node. 
* When I do specify `--node-ip-address`, the dashboard breaks, but I'm able to connect from a worker node
  * Except it doesn't fully work because the dashboard is broken, even when I pass `--include-dashboard=false` to the CLI and `include_dashboard=False` in the python script.

Again, not sure if this is the same issue. I might be doing something wrong in general here but I've tried to follow the guide pretty closely.
